File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/TVA.Phasors/IChannelCellCollection.vb
'*******************************************************************************************************
'  IChannelCellCollection.vb - Channel cell collection interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/18/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Namespace Phasors

    ''' <summary>This interface represents a protocol independent representation of a collection of any kind of data cell.</summary>
    <CLSCompliant(False)> _
    Public Interface IChannelCellCollection(Of T As IChannelCell)

        Inherits IChannelCollection(Of T)

    End Interface

End Namespace

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IChannelCellCollection.cs
//*******************************************************************************************************
//  IChannelCellCollection.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/18/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of a collection of <see cref="IChannelCell"/> objects.
    /// </summary>
    public interface IChannelCellCollection<T> : IChannelCollection<T> where T : IChannelCell
    {
        /// <summary>
        /// Gets flag that determines if the lengths of <see cref="IChannelCell"/> elements in this <see cref="IChannelCellCollection{T}"/> are constant.
        /// </summary>
        bool ConstantCellLength { get; }
    }
}
